python/lib/dcm2bids_imaging_pipeline_lib/base_pipeline.py

```diff
@@ -191,14 +192,10 @@ def determine_study_info(self):
          :rtype: dict
         """
 
-        cand_id = self.subject_id_dict['CandID']
-        visit_label = self.subject_id_dict['visitLabel']
-        patient_name = self.subject_id_dict['PatientName']
-
         # get the CenterID from the session table if the PSCID and visit label exists
         # and could be extracted from the database
-        if cand_id and visit_label:
-            self.session_obj.create_session_dict(cand_id, visit_label)
+        if self.subject.cand_id and self.subject.visit_label:
```

Comment:
It seems to me that this condition can never be false. `determine_study_info` is always called after `imaging_obj.determine_subject_ids`, which currently raises an exception if the dictionary is empty (or if the config is `None` in this PR, which is equivalent).
Exception raised here:
https://github.com/aces/Loris-MRI/blob/48231434b8ce9ed1f1f2091ced8673b3aabd2f2c/python/lib/imaging.py#L542
Only place where `determine_site_info` is called:
https://github.com/aces/Loris-MRI/blob/48231434b8ce9ed1f1f2091ced8673b3aabd2f2c/python/lib/dcm2bids_imaging_pipeline_lib/base_pipeline.py#L123

Reply:
If the PatientName is not matching the regex, you could end up with an empty dictionary. However, with your refactoring, I guess that instead you raise an exception so the code should probably be replaced here to handle the exception instead.